zenity: fix build for old systems and powerpc

barracuda156 opened the pull request:
#### Description

Fix build

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Triple-click and copy the next line and paste it into your shell. It will copy your OS and Xcode version to the clipboard. Paste it here replacing this section.
sh -c 'echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion) $(uname -m)"; xcode=$(xcodebuild -version 2>/dev/null); if [ $? == 0 ]; then echo "$(echo "$xcode" | awk '\''NR==1{x=$0}END{print x" "$NF}'\'')"; else echo "Command Line Tools $(pkgutil --pkg-info=com.apple.pkg.CLTools_Executables | awk '\''/version:/ {print $2}'\'')"; fi' | tee /dev/tty | pbcopy
-->
macOS 10.6
Xcode 3.2

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? <!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

barracuda156 commented:
@mascguy @ryandesign Should we just make `webkit` variant non-default for everyone?

ryandesign commented:
I don't know.

I do know that basing a decision on `configure.build_arch` will be wrong for universal builds, as I've said many times before.

barracuda156 commented:
> I do know that basing a decision on `configure.build_arch` will be wrong for universal builds, as I've said many times before.

@ryandesign Isn’t this case different though? We cannot build _one_ arch with an extra variant in a universal build. So it seems appropriate to disable webkit whenever at least one of target archs == powerpc. Or I miss something here?

reneeotten commented:
@ryandesign @barracuda156  this isn't really going anywhere... Any thoughts Ryan on Sergey's last comment? CN we find an acceptable/correct working solution here? If not, please close.

barracuda156 commented:
> @ryandesign @barracuda156 this isn't really going anywhere... Any thoughts Ryan on Sergey's last comment? CN we find an acceptable/correct working solution here? If not, please close.

Well, I could change that to `os.arch` after all, which would leave Rosetta broken, but no big deal, native ppc will be still fixed.

reneeotten commented:
please do as Ryan suggest and I'll merge it

barracuda156 commented:
> please do as Ryan suggest and I'll merge it

@reneeotten I do not know what to do about universal builds when this does not even use `muniversal` PG, however I dropped `configure.build_arch` now and replaced it with `os.arch`. That should be uncontroversial here. We know that on PowerPC WebKit does not build.
